src/nvim/api/vim.c

```diff
@@ -2149,6 +2164,31 @@ Dictionary nvim_eval_statusline(String str, Dict(eval_statusline) *opts, Error *
         fillchar = fillchar_status(&attr, wp);
       }
     }
+    if (use_statuscol) {
+      HlPriId line = { 0 };
+      HlPriId cul  = { 0 };
+      HlPriId num  = { 0 };
+      linenr_T lnum = (linenr_T)get_vim_var_nr(VV_LNUM);
+      int num_signs = buf_get_signattrs(wp->w_buffer, lnum, sattrs, &num, &line, &cul);
+      decor_redraw_signs(wp->w_buffer, lnum - 1, &num_signs, sattrs, &num, &line, &cul);
+
+      statuscol.sattrs = sattrs;
+      statuscol.foldinfo = fold_info(wp, lnum);
+      statuscol.use_cul = wp->w_p_cul && lnum == wp->w_cursorline
+                          && (wp->w_p_culopt_flags & CULOPT_NBR);
+      statuscol.sign_cul_id = statuscol.use_cul ? cul.hl_id : 0;
+      if (num.hl_id) {
+        statuscol.num_attr = num.hl_id;
```

Comment:
the field should be called `statuscol.num_hl_id` now that it stores an id

Reply:
I actually left this one to store attrs, at least for the drawing routines. It is not used in `build_stl_str_hl()` so we don't need it to be an id and this avoids resolving it multiple times in `get_statuscol_display_info()`.

I just abuse the field here(because here we do need the id) and resolve it in [`get_default_stl_hl()`](https://github.com/neovim/neovim/pull/22859/files#diff-fb7a3b0c5738d5de7da5beea244dae60458bf2fd476e5a6ca6dbbc52ed2ff32dR969) instead.

Reply:
Then it could be a separate `num_hl_id` field. we don't need to play attribute golf within a singleton struct.

Reply:
I added an `stc_hl_id` local to `nvim_eval_status()` instead. No need to extend the struct in this case.